include/seastar/net/tls.hh

```diff
@@ -330,9 +336,13 @@ namespace tls {
     ::seastar::socket socket(shared_ptr<certificate_credentials>, sstring name = {});
     /// @}
 
-    /** Wraps an existing connection in SSL/TLS. */
+    /**
+     * Wraps an existing connection in SSL/TLS.
+     *
+     * \param options Optional additional session configuration
+     */
     /// @{
-    future<connected_socket> wrap_client(shared_ptr<certificate_credentials>, connected_socket&&, sstring name = {});
+    future<connected_socket> wrap_client(shared_ptr<certificate_credentials>, connected_socket&&, sstring name = {}, std::optional<tls_options> options = std::nullopt);
```

Comment:
But why do we care? The caller sends options, either the default ones, or their own carefully crafted struct.

Reply:
No particular reason. I tend to think it more clear/easily eye-parsed when used. More or less just stylistic on my part. Feel free to override.

Reply:
I do want non-optional options, I think it's redunant. We use this style (without optional) everywhere.

Reply:
Sounds good. Will change to what you're suggesting.

Reply:
Done.

Reply:
`name` is the [Server Name Indication](https://www.cloudflare.com/en-gb/learning/ssl/what-is-sni/) which is a tls extension. I think it's recommended for clients to use it. To me moving `name` into the options would be a bit confusing since we'd need two overloads with different ways of specifying the SNI (one with a name param and another with options). Without an extra overload the change would break library users when they upgrade.

Reply:
The trick is to deprecate the old overload, then to remove it in a few months. Then we don't have some of the options as positional parameters, and some in option structs. See babac6de088ff8f4603fec0ca91baf3274225066 for an example.

Reply:
Makes sense. If we go down this route we'll have to change the other client APIs in the same way: `tls::connect` and `tls::socket` for the sake of consistency (new overloads and deprecate later). Not against doing it, but let's make sure we're in agreement first.

Reply:
@elcallio Added a new commit which changes the API as discussed.